Add EWS Infrastructure to support Monterey (252579)

robert-jenner opened the pull request:
#### eb60908303ab4874a8926527756e19b6ae6e4cdb
<pre>
Add EWS Infrastructure to support Monterey (252579)
<a href="https://bugs.webkit.org/show_bug.cgi?id=252579">https://bugs.webkit.org/show_bug.cgi?id=252579</a>
rdar://105684845

Reviewed by Ryan Haddad.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):

Canonical link: <a href="https://commits.webkit.org/261335@main">https://commits.webkit.org/261335@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd14c3e010f60cb5a3554f7c5a7bfb61faaa2fd6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109384 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18508 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42164 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/899 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118566 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113267 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19989 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9722 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/440 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115140 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14894 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98118 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101659 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96866 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29771 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43084 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11268 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31114 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84836 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11934 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8051 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/108150 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17301 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50717 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13650 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4302 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

ryanhaddad commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1228,7 +1228,7 @@ class CheckChangeRelevance(AnalyzeChange):
     group_to_paths_mapping = {
         'bindings': bindings_path_regexes,
-        'bigsur-release-build': big_sur_builder_path_regexes,
+        'monterey-release-build': big_sur_builder_path_regexes,
```
Rename `big_sur_builder_path_regexes` here and on line 1217.

ryanhaddad commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -1274,11 +1274,11 @@ def test_success_wpe(self):
     def test_failure(self):
         self.setupStep(CompileWebKit())
-        self.setProperty('fullPlatform', 'mac-bigsur')
+        self.setProperty('fullPlatform', 'mac-Monterey')
```
```suggestion
        self.setProperty('fullPlatform', 'mac-monterey')
```

ryanhaddad commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -1322,11 +1322,11 @@ def test_success(self):
     def test_failure(self):
         self.setupStep(CompileWebKitWithoutChange())
-        self.setProperty('fullPlatform', 'mac-bigsur')
+        self.setProperty('fullPlatform', 'mac-Monterey')
```
```suggestion
        self.setProperty('fullPlatform', 'mac-monterey')
```

ryanhaddad commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -2656,7 +2656,7 @@ def buildCommandKwargs(self, warnings):
         # work around log output delays on slower machines.
         # https://bugs.webkit.org/show_bug.cgi?id=247506: Only applies to Xcode 12.x.
-        if self.getProperty('fullPlatform') == 'mac-bigsur':
+        if self.getProperty('fullPlatform') == 'mac-monterey':
```
We'll need a follow up to remove this special case for the timeout, it only applied to Big Sur.

robert-jenner committed:
Add EWS Infrastructure to support Monterey (252579)
https://bugs.webkit.org/show_bug.cgi?id=252579
rdar://105684845

Reviewed by Ryan Haddad.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):

Canonical link: https://commits.webkit.org/261335@main
